engine/src/main/java/org/terasology/rendering/dag/nodes/ToneMappingNode.java

```diff
@@ -63,16 +63,16 @@ public ToneMappingNode(Context context) {
         screenGrabber = context.get(ScreenGrabber.class);
 
         DisplayResolutionDependentFBOs displayResolutionDependentFBOs = context.get(DisplayResolutionDependentFBOs.class);
-        requiresFBO(new FBOConfig(TONE_MAPPING_FBO, FULL_SCALE, FBO.Type.HDR), displayResolutionDependentFBOs);
-        addDesiredStateChange(new BindFbo(TONE_MAPPING_FBO, displayResolutionDependentFBOs));
-        addDesiredStateChange(new SetViewportToSizeOf(TONE_MAPPING_FBO, displayResolutionDependentFBOs));
+        FBO toneMappingFbo = requiresFBO(new FBOConfig(TONE_MAPPING_FBO, FULL_SCALE, FBO.Type.HDR), displayResolutionDependentFBOs);
+        addDesiredStateChange(new BindFbo(toneMappingFbo));
+        addDesiredStateChange(new SetViewportToSizeOf(toneMappingFbo));
 
         addDesiredStateChange(new EnableMaterial(TONE_MAPPING_MATERIAL));
 
         toneMappingMaterial = getMaterial(TONE_MAPPING_MATERIAL);
 
         int textureSlot = 0;
-        addDesiredStateChange(new SetInputTextureFromFbo(textureSlot, INITIAL_POST_FBO, ColorTexture, displayResolutionDependentFBOs, TONE_MAPPING_MATERIAL, "texScene"));
+        addDesiredStateChange(new SetInputTextureFromFbo(textureSlot, displayResolutionDependentFBOs.get(INITIAL_POST_FBO), ColorTexture, displayResolutionDependentFBOs, TONE_MAPPING_MATERIAL, "texScene"));
```

Comment:
Given the FBO manager is passed in anyway, why not leave an additional constructor that takes the FBO's URN as input, retrieves the FBO from the manager and simply passes it to the constructor being used here? It would shorten this line and others like it.